project/api/models.py

```diff
@@ -135,6 +135,28 @@ def save(self, *args, **kwargs):  # pylint: disable=W0221
         super(CachedData, self).save(*args, **kwargs)
 
 
+class City(models.Model):
+    """
+    Stores a city represented by a point on the map
+    """
+
+    wikidata_id = models.PositiveIntegerField()  # Excluding the Q
```

Comment:
Is this sufficient to identify the city or do we want a name field as well?

Reply:
Sadly, In wikidata some signficant cities does not have exact inception date (e.g. London or Moscow)

Reply:
I would put the city name here too. Or else, we need to make n wikidata requests from the frontend. Thoughts @MiklerGM?

Reply:
Perhaps it might be a good idea to do the same for CachedData objects as well?

Reply:
It is not that easy with city names,  they are not  constant. City can ba renamed. Check [Saint-Petersburg](https ://www.wikidata.org/wiki/Q656) for example.
I want the city label in this table. Mb we should create duplicate entities with different labels and dates?

Btw, we should check Republic of Macedonia after renaming

Reply:
Good point. I would go for multiple rows for different names, there's currently no uniqueness constraint on the wid.

Reply:
AFAIK wikidata does not have any constraints on dates overlapping and missing dates.
I might be wrong, but afair
city_label can be like this. 
A from 1717 to 1719
B from 1719
C from 1790 to 1880
D from 1879

Reply:
> Perhaps it might be a good idea to do the same for CachedData objects as well?

I think it's less important, because those names are fetched on mouse hover.

> city_label can be like this.
A from 1717 to 1719
B from 1719
C from 1790 to 1880
D from 1879

Those would be 4 different wikidata items, correct?

Reply:
I am worrying about if it is a single wikidata item, and different `official name` properties. That would result two names for same point at the same time.

I am not sure it is common, but dates overlapping is possible in different properties.